client/components/RegistrationModal/RegistrationModal.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useState } from "react";
+import Modal from "../Modal/Modal"; // Adjust the import path as needed
+import { ModalProps } from "../Modal/types";
+
+import styles from "./RegistrationModal.module.css";
+import SelectMenu from "../SelectMenu/SelectMenu";
+import states from "./states.json";
+import RadioSelect from "../RadioSelect/RadioSelect";
+import { useGetAllGroups } from "../../api/groups/query";
+import { useRegisterUser } from "../../api/users/mutation";
+import { GroupType } from "../../api/groups/types";
+import { RegisterUser } from "../../api/users/types";
+
+//TODO: Arturo uses Form modules css to adjust register data, make copy as needed
+import { useAuth0 } from "@auth0/auth0-react";
+
+// Extend ModalProps to include the onRegistrationComplete callback
+interface RegisterModalProps extends ModalProps {
+  onRegistrationComplete: () => void; // Callback for when registration is complete
+}
+
+const RegisterModal: React.FC<RegisterModalProps> = ({
+  open,
+  onOpenChange,
+  onRegistrationComplete,
+}) => {
+  const [page, setPage] = useState<number>(1);
+  const [isExistingOrg, setisExistingOrg] = useState<string>("No");
+  const [org, setOrg] = useState<string>("");
+  const [address, setAddress] = useState<string>("");
+  const [city, setCity] = useState<string>("");
+  const [state, setState] = useState<string>("");
+  const [zipCode, setZipCode] = useState<string>("");
+  const [selectedOrg, setSelectedOrg] = useState<string>("");
```

Comment:
TS can already infer its type so it won't be needed. Also since  this logic is dependent on submitting a form consider using [useReducer](https://react.dev/reference/react/useReducer)


```suggestion
  const [page, setPage] = useState(1);
  const [isExistingOrg, setisExistingOrg] = useState("No");
  const [org, setOrg] = useState("");
  const [address, setAddress] = useState("");
  const [city, setCity] = useState("");
  const [state, setState] = useState("");
  const [zipCode, setZipCode] = useState("");
  const [selectedOrg, setSelectedOrg] = useState("");
```

Reply:
Done! Thank you for the knowledge! Also implemented `useReducer` alternative to form ingestion!